File: src/widgets/CollapsibleWidget.cpp
#include "CollapsibleWidget.h"

#include <QPushButton>
#include <QVBoxLayout>

CollapsibleWidget::CollapsibleWidget(const QString& title, QWidget* contentWidget, QWidget* parent)
    : QWidget(parent), contentWidget_(contentWidget)
{
    QVBoxLayout* mainLayout = new QVBoxLayout(this);
    mainLayout->setSpacing(0);
    mainLayout->setContentsMargins(0,0,0,0);

    // Header
    headerButton_ = new QPushButton(title);
    headerButton_->setCheckable(true);
    headerButton_->setChecked(true);
    headerButton_->setObjectName("headerButton");
    mainLayout->addWidget(headerButton_);

    // Contenu
    mainLayout->addWidget(contentWidget_);
    contentWidget_->setVisible(true);

    connect(headerButton_, &QPushButton::toggled, contentWidget_, &QWidget::setVisible);
}

void CollapsibleWidget::setCollapsed(bool collapsed)
{
    headerButton_->setChecked(!collapsed);
    contentWidget_->setVisible(!collapsed);
}

bool CollapsibleWidget::isCollapsed() const
{
    return !headerButton_->isChecked();
}

File: viewer/widgets/collapsibleWidget.py
from PyQt5.QtWidgets import QWidget, QPushButton, QVBoxLayout

class CollapsibleWidget(QWidget):
    def __init__(self, title: str, content_widget: QWidget, parent: QWidget = None):
        super().__init__(parent)
        self.content_widget = content_widget

        # Layout principal
        main_layout = QVBoxLayout(self)
        main_layout.setSpacing(0)
        main_layout.setContentsMargins(0, 0, 0, 0)

        # Header
        self.header_button = QPushButton(title)
        self.header_button.setCheckable(True)
        self.header_button.setChecked(True)
        self.header_button.setObjectName("headerButton")
        main_layout.addWidget(self.header_button)

        # Contenu
        main_layout.addWidget(self.content_widget)
        self.content_widget.setVisible(True)

        # Connexion toggle
        self.header_button.toggled.connect(self.content_widget.setVisible)

    def setCollapsed(self, collapsed: bool):
        """Plie ou déplie le widget"""
        self.header_button.setChecked(not collapsed)
        self.content_widget.setVisible(not collapsed)

    def isCollapsed(self) -> bool:
        """Retourne True si le widget est plié"""
        return not self.header_button.isChecked()
